alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt

```diff
@@ -2379,7 +2380,9 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
         val trigger = randomDocumentLevelTrigger(condition = ALWAYS_RUN)
         var monitor = randomDocumentLevelMonitor(
             inputs = listOf(docLevelInput),
-            triggers = listOf(trigger)
+            triggers = listOf(trigger),
+            dataSources = DataSources(),
+            owner = "alerting"
```

Comment:
plz add 2 new tests-
1. scenario where delete query index flag is false and monitor execution fails due to mapping conflict on alias rollover
2.  scenario where delete query index flag is true and monitor execution succeeds

Reply:
in first test plz also update monitor with flag false to true and do rollover again and this time assert monitor execution succeeeds